src/UiPath.Workflow.Runtime/ActivityContext.cs

```diff
@@ -86,7 +86,11 @@ public WorkflowDataContext DataContext
         }
     }
```

Comment:
this feature can be done in an extension class; it's better not to mess with core classes Activity/ActivityInstance/ActivityContext if not necessary.

Reply:
My suggestion in a separate file, including some naming changes is:
```
+namespace UiPath.Workflow.Runtime.ParallelTracking
+{
+    public static class ParallelTrackingExtensions
+    {
+        public const string BranchIdPropertyName = "__ParallelBranchId";
+
+        public static ActivityInstance MarkNewParallelBranch(this ActivityInstance instance)
+        {
+            var properties = new ExecutionProperties(null, instance, instance.PropertyManager);
+            var parentId = properties.Find(BranchIdPropertyName) as string;
+            properties.Add(BranchIdPropertyName, $"{parentId}.{Guid.NewGuid():N}", true, false);
+            return instance;
+        }
+
+        public static string GetCurrentParallelBranchId(this ActivityInstance instance) =>
+            instance.PropertyManager?.GetPropertyAtCurrentScope(BranchIdPropertyName) as string;
+    }
```
we can include the extensions on ActivityContext also.

Reply:
done